naviguide.cc

```diff
@@ -67,36 +67,52 @@ typedef struct {
 /* Global variables */
 
 static gbfile* file_in, *file_out;
-static uint16_t nof_wp;
+static uint16_t nof_wp = 0;
 static route_head* rte_head;
 static ng_file_header_t ng_file_header;
 static ng_wp_no_comment_t WPNC;
 static ng_next_wp_t ng_next_wp;
 static char strComment[101];
 
-/* Process options */
-/* wp - process only waypoints */
-/* rte - process as route */
-/* wprte - Process waypoints and route */
-static char* process = NULL;
-static char* reorder = NULL;
-static int process_rte = 1;
-static int reorder_wp = 0;
+/*
+    File reading options
+    wp - read file as waypoints
+    rte - read file as route
+    trk - read file as track
 
-static char temp_short_name[5];
+ */
+static char* read_as = NULL;
+gpsdata_type read_type = unknown_gpsdata;
 
+/*
+    File writing options:
+    First non-empty data type, as a string with the letters:
+    w - write waypoints to the file, if any.
+    r - write routes to the file, if any.
+    t - write tracks to the file, if any.
 
+ */
+static char* write_as = NULL;
+gpsdata_type write_type = unknown_gpsdata;
+
+static char* reorder = NULL;
+static int reorder_wp = 0;
 
+static char temp_short_name[5];
 
 /* Forward declarations */
 static void ng_read_file_header();
 
 static
 arglist_t ng_args[] = {
   {
-    "output", &process, "'wp' - Create waypoint file , 'rte' - Create route file",
+    "output", &read_as, "'wp' - Read file as waypoints, 'rte' - Read file as route, 'trk' - Read file as track",
     "rte", ARGTYPE_STRING, ARG_NOMINMAX, nullptr
   },
+  {
+    "input", &write_as, "Select the source type for writing, with optional priority.",
```

Comment:
This probably wasn't well though through. Let's try to identify the files on read and just do the right thing without making users select anything at all here.